### PR TITLE
Test that transient settings beat persistent ones

### PR DESCRIPTION
**Pull request by DaveCTurner**
Transient settings override persistent settings, but in fact all of the tests
that run as part of `:server:test` and `:server:integTest` will pass if the
precedence is changed to be the other way round. This change adds a test that
verifies the precedence is as documented.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataTests.java [hunk 1]
```diff
@@ -794,4 +794,12 @@ public static void assertMultiField(Map<String, Object> properties, String field
             "    }\n" +
             "  }\n" +
             "}";
+
+    public void testTransientSettingsOverridePersistentSettings() {
+        MetaData metaData = MetaData.builder()
+            .persistentSettings(Settings.builder().put("key", "persistent-value").build())
+            .transientSettings(Settings.builder().put("key", "transient-value").build()).build();
+
+        assertThat(metaData.settings().get("key"), equalTo("transient-value"));
```
Comment (jasontedor) [inline server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataTests.java]: I did not check but I think it should be possible to test this with a setting object instead doing a raw get on the settings instance; that is `Settings#get(String)` versus `Setting#get(settings)`. Would you please write the test that way? It is more realistic.

**Review by jasontedor — COMMENTED** [review]
Thanks for adding this test, it's really good to have tests for fundamental assumptions such as this. 🙈

I left a request.

File: server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataTests.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/metadata/MetaDataTests.java]: Yes, this was possible. I pushed 1ec65d0.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**